nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -516,38 +516,53 @@ sub addEntry {
     $conf .= "}\n\n";
 }
 
+sub addGeneration {
+    my ($name, $nameSuffix, $path, $options, $current) = @_;
 
-# Add default entries.
-$conf .= "$extraEntries\n" if $extraEntriesBeforeNixOS;
+    # Do not search for grand children
+    my @links = sort (glob "$path/specialisation/*");
 
-addEntry("@distroName@ - Default", $defaultConfig, $entryOptions, 1);
+    if (scalar(@links) != 0) {
+        $conf .= "submenu \"Submenu for $name$nameSuffix\" --class submenu {\n";
```

Comment:
I'm not fond of adding "Submenu for" to every generations. Though I agree there needs to be a way to determine whether or not this is going to boot a system of go into a submenu.

The phrasing could be "Specialisations for ...", but still a bit cumbersome to me.

Maybe a symbol? Unicode would work on default configs, but not if the user uses other fonts. So let's keep it simple :(.

```suggestion
        $conf .= "submenu \"$name$nameSuffix >\" --class submenu {\n";
```

Not sure what else other than a `>` at the end could work here.

Reply:
There's still the question about how to handle these...

In my opinion the "Submenu for" prefix becomes extra-redundant given that in my experience, it is quite rare to have only a couple generations with specialisations... In my use-case, they are used for a long time.

But I do agree that there needs to be a signal helping the user know "this one is a submenu, this one is not".

Any thoughts?